Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,433 @@
+        if !configuration.noList {
+            let output = try self.listEndpoints(samExecutablePath: samExecutablePath,
+                                                samDeploymentDescriptorFilePath: samDeploymentDescriptorFilePath,
+                                                stackName : configuration.stackName,
+                                                verboseLogging: configuration.verboseLogging)
+            print(output)
+        }
+    }
+    
+    private func generateDeploymentDescriptor(projectDirectory: Path,
+                                              buildConfiguration: PackageManager.BuildConfiguration,
+                                              swiftExecutable: Path,
+                                              samDeploymentDescriptorFilePath: String,
+                                              archivePath: String,
+                                              force: Bool,
+                                              verboseLogging: Bool) throws {
+        print("-------------------------------------------------------------------------")
+        print("Generating SAM deployment descriptor")
+        print("-------------------------------------------------------------------------")
+        
+        //
+        // Build and run the Deploy.swift package description
+        // this generates the SAM deployment descriptor
+        //
+        let deploymentDescriptorFileName = "Deploy.swift"
+        let deploymentDescriptorFilePath = "\(projectDirectory)/\(deploymentDescriptorFileName)"
+        let sharedLibraryName = "AWSLambdaDeploymentDescriptor" // provided by the swift lambda runtime
+        
+        // Check if Deploy.swift exists. Stop when it does not exist.
+        guard FileManager.default.fileExists(atPath: deploymentDescriptorFilePath) else {
+            print("`Deploy.Swift` file not found in directory \(projectDirectory)")
+            throw DeployerPluginError.deployswiftDoesNotExist
+        }
+        
+        do {
+            let cmd = [
+                swiftExecutable.string,
+                "-L \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-I \(projectDirectory)/.build/\(buildConfiguration)/",
+                "-l\(sharedLibraryName)",
+                deploymentDescriptorFilePath,
+                "--archive-path", archivePath
+            ]
+            let helperCmd = cmd.joined(separator: " \\\n")
+            
+            if verboseLogging {
+                print("-------------------------------------------------------------------------")
+                print("Swift compile and run Deploy.swift")
+                print("-------------------------------------------------------------------------")
+                print("Swift command:\n\n\(helperCmd)\n")
+            }
+            
+            // create and execute a plugin helper to run the "swift" command
+            let helperFilePath = "\(projectDirectory)/compile.sh"
```

Comment:
should this go to a temp directory so not to "pollute" the project directory?

Reply:
done